sap-xmake-build.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash -l
+echo "Inside file -- sap-xmake-build.sh --"
+
+# find this script and establish base directory
+curr_dir=`dirname $0`
+MY_DIR=`cd $curr_dir; pwd`
+echo "[INFO] Executing in ${MY_DIR}"
+
+PACKAGE_NAME=${PACKAGE_NAME:-"scala-pickling"}
+PACKAGE_BRANCH=${PACKAGE_BRANCH:-"0.11.x"}
+BUILD_BRANCH=${BUILD_BRANCH:-"v0.11.x-M1_2.11"}
+ALTISCALE_RELEASE="${ALTISCALE_RELEASE:-5.0.0}"
+
+rm -rf install-dir-*
+rm -rf rpm-dir-*
+
+echo "ok - $(whoami) user is going to build the project"
+echo "ok - fpm is located at $(which fpm)"
+echo "ok - PATH=$PATH"
+echo "ok - BUILD_BRANCH=$BUILD_BRANCH"
+echo "ok - PACKAGE_BRANCH=$PACKAGE_BRANCH"
+
+ret=0
+export DATE_STRING=${DATE_STRING:-$(date -u +%Y%m%d%H%M)}
+RPM_INSTALL_DIR=${MY_DIR}/picklingrpmbuild
+echo "RPM_INSTALL_DIR : $RPM_INSTALL_DIR"
+pushd ${MY_DIR}
+
+export RPM_NAME=`echo scala-pickling-${BUILD_BRANCH}`
+export RPM_DESCRIPTION="scala-pickling library ${BUILD_BRANCH}"
+echo "RPM_NAME : $RPM_NAME"
+echo "RPM_DESCRIPTION : $RPM_DESCRIPTION"
+echo "Current directory : ${pwd}"
```

Comment:
this is not printing anything.  The correct syntax is $(pwd), not brace.
Also, please clean up the debug printout as most echo are not needed.

Reply:
Winnie, I use lots of echos myself because it has been very useful while following the build output

Reply:
i'm not against echo, but some are redundant here.

Reply:
also the ${pwd} syntax is wrong, it doesn't print anything

Reply:
Yes it should be $(pwd)

Reply:
rule of thumbs, echo is ok as long as the output isn't lengthy and polluted which means.... lots of logs to look at >.< I sometimes add a shell function with debug flag, but it could be overkilled.